Add xcframework code signing

jkmassel opened the pull request:
Adds code signing to the release and debug XCFrameworks. This allows consumer projects to ensure the binary dependency hasn't been tampered with.

jkmassel committed:
Add xcframework code signing

crazytonyli commented on fastlane/Fastfile:
```diff
@@ -396,6 +411,27 @@ lane :generate_fluent_file_from_po do |file_path:|
+    team_id: APPLE_TEAM_ID,
+    api_key: app_store_connect_api_key,
+    type: 'development',
+    certificate_id: 'Apple Development: Created via API (886NX39KP6)',
```
What do you think about using the "Apple Distribution" one instead? I feel like it's appropriate to use binaries that are signed with "Apple Distribution" on all builds, but "Apple Development" builds may not be suitable for release builds.

crazytonyli commented on fastlane/Fastfile:
```diff
@@ -24,6 +24,21 @@ PROJECT_NAME = 'wordpress-rs'
 
+# Code Signing
+APPLE_TEAM_ID = 'PZYM8XX95Q'
+APPLE_BUNDLE_IDENTIFIER = 'com.automattic.hostmgr'
```
If I understand it correctly, this bundle id is only for fastlane to download the certificates and things. If that's the case, can we move it into the `set_up_signing` lane, to avoid the potential confusion that `com.automattic.hostmgr` is the "bundle id" of the wordpress-rs xcframework?

mokagio committed:
Remove double blank line

mokagio committed:
Use distribution certificate for XCFramework code signing

mokagio committed:
Address a RuboCop string formatting violation

mokagio committed:
Add `setup_ci` call to Fastlane

mokagio committed:
Use `set_up_signing_release` in CI

mokagio committed:
Add more CI logs headers

Otherwise, looking at the logs without expanding them would make you
think the build is stuck on sync_code_signing.

See
https://buildkite.com/automattic/wordpress-rs/builds/4084/steps/canvas?sid=019a531b-f04b-4764-9cf7-673e189f3f26

mokagio commented:
Sorry, I meant to create a dedicated branch for the distribution certificate (tracked in https://linear.app/a8c/issue/AINFRA-1477/make-a-release-mode-code-signing-certificate) but I pushed without realizing I was on this branch...

mokagio commented:
I know how to fix the build failures caused by the change in the artifact path... but my Docker is bricked and that is blocking my shell from entering into the this repo's folder -.-'

mokagio committed:
Update `download-xcframework.sh` after changing ZIP location

mokagio committed:
Temporarily also download XCFramework ZIP to target

mokagio committed:
Add `popd` following `pushd`

mokagio commented on Makefile:
```diff
@@ -137,6 +139,9 @@ xcframework-package: xcframework-all
 	swift package compute-checksum libwordpressFFI.xcframework.zip | tee libwordpressFFI.xcframework.zip.checksum.txt
 
+xcframework-sign:
+	codesign --timestamp -v --sign "${certificate_name_release}" target/libwordpressFFI.xcframework
```
Working good. See https://buildkite.com/automattic/wordpress-rs/builds/4086/steps/canvas?sid=019a538a-c172-4ce7-9a98-a586299292e5#019a538a-c1c1-4c09-9d8d-36437a2435fb/704-2883

<img width="1962" height="130" alt="image" src="https://github.com/user-attachments/assets/9e456040-be91-45da-8d29-74fec0b21ac7" />

mokagio commented on .buildkite/download-xcframework.sh:
```diff
@@ -3,7 +3,15 @@
+pushd target
+buildkite-agent artifact download libwordpressFFI.xcframework.zip . --step "xcframework"
+unzip libwordpressFFI.xcframework.zip -d .
+rm libwordpressFFI.xcframework.zip
```
@jkmassel @crazytonyli This PR changed the ZIP destination from `target/` to `./` in https://github.com/Automattic/wordpress-rs/pull/966/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53L74

I thought it would be enough to update the download call at the start of the script (see above) but there evidently are other parts of the automation that expect the file to be in `target/`, because to make the build pass I had to add this additional ZIP expansion in the `target/` folder.

I'm happy to help tidy this up. Alternatively, we could restore the ZIP-to-target-folder behavior and leave it at that. Up to you.

crazytonyli replied:
Looks like `make xcframework-package` was introduced to replace the `zip` command. Maybe we can update `xcframework-package` to put the zip file at `target/`, and you don't need the new script here to unzip twice?

mokagio replied:
I did that and now there's other stuff that fails...

I gotta go offline for a bit but I'll get to the bottom of this later/tomorrow

crazytonyli replied:
Was the issue around the unzipped content not in the `target` directory? It's probably because the folder structures of the `zip`-ed and `ditto`-ed zip files are different. 😥

mokagio committed:
Update `Makefile` to package XCFramework in target folder

See @crazytonyli's suggestion
https://github.com/Automattic/wordpress-rs/pull/966#discussion_r2496397070

mokagio committed:
Add required mkdir -p to download script

mokagio committed:
Update XCFramework path to reflect restored `target/` dir

mokagio committed:
Fix download script one last time

mokagio committed:
Unzip in `./target/` (?)
